### ISSUE TITLE
Sunshine creates new gamepad on button press

### ISSUE THREAD

**Issue by nachoz12341** [LizardByte/Sunshine#1347]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your issue described in the documentation?

- [X] I have read the documentation

### Is your issue present in the nightly release?

- [X] This issue is present in the nightly release

### Describe the Bug

Streaming from my pc to my android device, sometimes it will stop recognizing gamepad input instead opting to create a new virtual gamepad everytime I press a button. Screen shot attached below. Logs show multiple vigem errors
[2023:06:02:11:30:38]: Error: Couldn't add Gamepad to ViGEm connection [E0000007]

Streaming pc:
i7 7700
16gb ddr 1833mhz
rtx 3070
1tb nvme ssd
first observed with sunshine Version 0.19.1.d70d084f9fbb4e0150977a89d94937418a3ccf9c

Android device:
nvidia shield portable
nvidia tegra 4
2gb ram
android 5.1
using latest moonlight on google play v11.0

![image](https://github.com/LizardByte/Sunshine/assets/40530732/7cb2c576-a6fe-458e-943b-94d9e14914f7)

### Expected Behavior

Single virtual gamepad for one hardware gamepad on streaming device

### Additional Context

Doesn't always occur. It seems to happen most with games not on steam.

### Host Operating System

Windows

### Operating System Version

Windows 10

### Architecture

64 bit

### Sunshine commit or version

Version 0.19.1.d70d084f9fbb4e0150977a89d94937418a3ccf9c

### Package

Windows - installer

### GPU Type

Nvidia

### GPU Model

RTX 3070

### GPU Driver/Mesa Version

516.94

### Config

```Shell
Sunshine.conf is empty
```

### Relevant log output

```shell
[2023:06:01:23:28:59]: Info: Sunshine version: 0.19.1.d70d084f9fbb4e0150977a89d94937418a3ccf9c
[2023:06:01:23:28:59]: Info: Compiling shaders...
[2023:06:01:23:28:59]: Info: Compiled shaders
[2023:06:01:23:28:59]: Info: // Testing for available encoders, this may generate errors. You can safely ignore those errors. //
[2023:06:01:23:28:59]: Info: Trying encoder [nvenc]
[2023:06:01:23:28:59]: Info: System tray created
[2023:06:01:23:28:59]: Info: ddprobe.exe [1] [] returned: 0x80070005
[2023:06:01:23:28:59]: Info: Set GPU preference: 1
[2023:06:01:23:28:59]: Info: 
Device Description : NVIDIA GeForce RTX 3070
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002484
Device Video Mem   : 8043 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 8154 MiB
Feature Level      : 0x0000B100
Capture size       : 1280x720
Offset             : 0x0
Virtual Desktop    : 1280x720
[2023:06:01:23:28:59]: Info: Desktop resolution [1280x720]
[2023:06:01:23:28:59]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:06:01:23:28:59]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.639648,0.339844]
Green Primary      : [0.299805,0.69043]
Blue Primary       : [0.137695,0.0380859]
White Point        : [0.282227,0.296875]
Min Luminance      : 0.01 nits
Max Luminance      : 1499 nits
Max Full Luminance : 799 nits
[2023:06:01:23:28:59]: Info: SDR color coding [Rec. 601]
[2023:06:01:23:28:59]: Info: Color range: [JPEG]
[2023:06:01:23:29:00]: Info: 
Device Description : NVIDIA GeForce RTX 3070
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002484
Device Video Mem   : 8043 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 8154 MiB
Feature Level      : 0x0000B100
Capture size       : 1280x720
Offset             : 0x0
Virtual Desktop    : 1280x720
[2023:06:01:23:29:00]: Info: Desktop resolution [1280x720]
[2023:06:01:23:29:00]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:06:01:23:29:00]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.639648,0.339844]
Green Primary      : [0.299805,0.69043]
Blue Primary       : [0.137695,0.0380859]
White Point        : [0.282227,0.296875]
Min Luminance      : 0.01 nits
Max Luminance      : 1499 nits
Max Full Luminance : 799 nits
[2023:06:01:23:29:00]: Info: SDR color coding [Rec. 601]
[2023:06:01:23:29:00]: Info: Color range: [JPEG]
[2023:06:01:23:29:00]: Info: 
Device Description : NVIDIA GeForce RTX 3070
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002484
Device Video Mem   : 8043 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 8154 MiB
Feature Level      : 0x0000B100
Capture size       : 1280x720
Offset             : 0x0
Virtual Desktop    : 1280x720
[2023:06:01:23:29:00]: Info: Desktop resolution [1280x720]
[2023:06:01:23:29:00]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:06:01:23:29:00]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.639648,0.339844]
Green Primary      : [0.299805,0.69043]
Blue Primary       : [0.137695,0.0380859]
White Point        : [0.282227,0.296875]
Min Luminance      : 0.01 nits
Max Luminance      : 1499 nits
Max Full Luminance : 799 nits
[2023:06:01:23:29:00]: Info: SDR color coding [Rec. 601]
[2023:06:01:23:29:00]: Info: Color range: [JPEG]
[2023:06:01:23:29:00]: Info: 
Device Description : NVIDIA GeForce RTX 3070
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002484
Device Video Mem   : 8043 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 8154 MiB
Feature Level      : 0x0000B100
Capture size       : 1280x720
Offset             : 0x0
Virtual Desktop    : 1280x720
[2023:06:01:23:29:00]: Info: Desktop resolution [1280x720]
[2023:06:01:23:29:00]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:06:01:23:29:00]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.639648,0.339844]
Green Primary      : [0.299805,0.69043]
Blue Primary       : [0.137695,0.0380859]
White Point        : [0.282227,0.296875]
Min Luminance      : 0.01 nits
Max Luminance      : 1499 nits
Max Full Luminance : 799 nits
[2023:06:01:23:29:00]: Info: SDR color coding [Rec. 601]
[2023:06:01:23:29:00]: Info: Color range: [JPEG]
[2023:06:01:23:29:01]: Info: 
Device Description : NVIDIA GeForce RTX 3070
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002484
Device Video Mem   : 8043 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 8154 MiB
Feature Level      : 0x0000B100
Capture size       : 1280x720
Offset             : 0x0
Virtual Desktop    : 1280x720
[2023:06:01:23:29:01]: Info: Desktop resolution [1280x720]
[2023:06:01:23:29:01]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:06:01:23:29:01]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.639648,0.339844]
Green Primary      : [0.299805,0.69043]
Blue Primary       : [0.137695,0.0380859]
White Point        : [0.282227,0.296875]
Min Luminance      : 0.01 nits
Max Luminance      : 1499 nits
Max Full Luminance : 799 nits
[2023:06:01:23:29:01]: Info: SDR color coding [Rec. 601]
[2023:06:01:23:29:01]: Info: Color range: [JPEG]
[2023:06:01:23:29:01]: Info: 
Device Description : NVIDIA GeForce RTX 3070
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002484
Device Video Mem   : 8043 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 8154 MiB
Feature Level      : 0x0000B100
Capture size       : 1280x720
Offset             : 0x0
Virtual Desktop    : 1280x720
[2023:06:01:23:29:01]: Info: Desktop resolution [1280x720]
[2023:06:01:23:29:01]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:06:01:23:29:01]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.639648,0.339844]
Green Primary      : [0.299805,0.69043]
Blue Primary       : [0.137695,0.0380859]
White Point        : [0.282227,0.296875]
Min Luminance      : 0.01 nits
Max Luminance      : 1499 nits
Max Full Luminance : 799 nits
[2023:06:01:23:29:01]: Info: SDR color coding [Rec. 709]
[2023:06:01:23:29:01]: Info: Color range: [JPEG]
[2023:06:01:23:29:01]: Error: Could not open codec [h264_nvenc]: Function not implemented
[2023:06:01:23:29:01]: Error: DuplicateOutput() test failed [0x80070005]
[2023:06:01:23:29:01]: Error: Failed to locate an output device
[2023:06:01:23:29:02]: Info: 
Device Description : NVIDIA GeForce RTX 3070
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002484
Device Video Mem   : 8043 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 8154 MiB
Feature Level      : 0x0000B100
Capture size       : 1280x720
Offset             : 0x0
Virtual Desktop    : 1280x720
[2023:06:01:23:29:02]: Info: Desktop resolution [1280x720]
[2023:06:01:23:29:02]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:06:01:23:29:02]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.639648,0.339844]
Green Primary      : [0.299805,0.69043]
Blue Primary       : [0.137695,0.0380859]
White Point        : [0.282227,0.296875]
Min Luminance      : 0.01 nits
Max Luminance      : 1499 nits
Max Full Luminance : 799 nits
[2023:06:01:23:29:02]: Info: SDR color coding [Rec. 709]
[2023:06:01:23:29:02]: Info: Color range: [JPEG]
[2023:06:01:23:29:02]: Info: 
Device Description : NVIDIA GeForce RTX 3070
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002484
Device Video Mem   : 8043 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 8154 MiB
Feature Level      : 0x0000B100
Capture size       : 1280x720
Offset             : 0x0
Virtual Desktop    : 1280x720
[2023:06:01:23:29:02]: Info: Desktop resolution [1280x720]
[2023:06:01:23:29:02]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:06:01:23:29:02]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.639648,0.339844]
Green Primary      : [0.299805,0.69043]
Blue Primary       : [0.137695,0.0380859]
White Point        : [0.282227,0.296875]
Min Luminance      : 0.01 nits
Max Luminance      : 1499 nits
Max Full Luminance : 799 nits
[2023:06:01:23:29:02]: Info: SDR color coding [Rec. 601]
[2023:06:01:23:29:02]: Info: Color range: [JPEG]
[2023:06:01:23:29:03]: Info: 
Device Description : NVIDIA GeForce RTX 3070
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002484
Device Video Mem   : 8043 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 8154 MiB
Feature Level      : 0x0000B100
Capture size       : 1280x720
Offset             : 0x0
Virtual Desktop    : 1280x720
[2023:06:01:23:29:03]: Info: Desktop resolution [1280x720]
[2023:06:01:23:29:03]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:06:01:23:29:03]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.639648,0.339844]
Green Primary      : [0.299805,0.69043]
Blue Primary       : [0.137695,0.0380859]
White Point        : [0.282227,0.296875]
Min Luminance      : 0.01 nits
Max Luminance      : 1499 nits
Max Full Luminance : 799 nits
[2023:06:01:23:29:03]: Info: SDR color coding [Rec. 601]
[2023:06:01:23:29:03]: Info: Color range: [JPEG]
[2023:06:01:23:29:03]: Warning: nvenc: hevc: replacing nalu prefix data
[2023:06:01:23:29:03]: Info: 
[2023:06:01:23:29:03]: Info: // Ignore any errors mentioned above, they are not relevant. //
[2023:06:01:23:29:03]: Info: 
[2023:06:01:23:29:03]: Info: Found encoder nvenc: [h264_nvenc, hevc_nvenc]
[2023:06:01:23:29:03]: Info: Configuration UI available at [https://localhost:47990]
[2023:06:01:23:29:04]: Info: Registered Sunshine mDNS service
[2023:06:02:00:07:15]: Info: Spawning [steam steam://open/bigpicture] in ["C:\Program Files (x86)\Steam"]
[2023:06:02:00:07:15]: Info: steam steam://open/bigpicture running with PID 17052
[2023:06:02:00:07:15]: Info: Executing [Desktop]
[2023:06:02:00:07:15]: Info: CLIENT CONNECTED
[2023:06:02:00:07:16]: Info: 
Device Description : NVIDIA GeForce RTX 3070
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002484
Device Video Mem   : 8043 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 8154 MiB
Feature Level      : 0x0000B100
Capture size       : 1280x720
Offset             : 0x0
Virtual Desktop    : 1280x720
[2023:06:02:00:07:16]: Info: Desktop resolution [1280x720]
[2023:06:02:00:07:16]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:06:02:00:07:16]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.639648,0.339844]
Green Primary      : [0.299805,0.69043]
Blue Primary       : [0.137695,0.0380859]
White Point        : [0.282227,0.296875]
Min Luminance      : 0.01 nits
Max Luminance      : 1499 nits
Max Full Luminance : 799 nits
[2023:06:02:00:07:16]: Info: Capture format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:06:02:00:07:16]: Info: SDR color coding [Rec. 601]
[2023:06:02:00:07:16]: Info: Color range: [MPEG]
[2023:06:02:00:21:32]: Info: 
Device Description : NVIDIA GeForce RTX 3070
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002484
Device Video Mem   : 8043 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 8154 MiB
Feature Level      : 0x0000B100
Capture size       : 1280x720
Offset             : 0x0
Virtual Desktop    : 1280x720
[2023:06:02:00:21:32]: Info: Desktop resolution [1280x720]
[2023:06:02:00:21:32]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:06:02:00:21:32]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.639648,0.339844]
Green Primary      : [0.299805,0.69043]
Blue Primary       : [0.137695,0.0380859]
White Point        : [0.282227,0.296875]
Min Luminance      : 0.01 nits
Max Luminance      : 1499 nits
Max Full Luminance : 799 nits
[2023:06:02:00:21:33]: Info: Capture format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:06:02:00:21:33]: Info: SDR color coding [Rec. 601]
[2023:06:02:00:21:33]: Info: Color range: [MPEG]
[2023:06:02:00:22:54]: Info: CLIENT DISCONNECTED
[2023:06:02:00:23:11]: Info: CLIENT CONNECTED
[2023:06:02:00:23:11]: Info: 
Device Description : NVIDIA GeForce RTX 3070
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002484
Device Video Mem   : 8043 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 8154 MiB
Feature Level      : 0x0000B100
Capture size       : 1280x720
Offset             : 0x0
Virtual Desktop    : 1280x720
[2023:06:02:00:23:11]: Info: Desktop resolution [1280x720]
[2023:06:02:00:23:11]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:06:02:00:23:11]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.639648,0.339844]
Green Primary      : [0.299805,0.69043]
Blue Primary       : [0.137695,0.0380859]
White Point        : [0.282227,0.296875]
Min Luminance      : 0.01 nits
Max Luminance      : 1499 nits
Max Full Luminance : 799 nits
[2023:06:02:00:23:11]: Info: Capture format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:06:02:00:23:11]: Info: SDR color coding [Rec. 601]
[2023:06:02:00:23:11]: Info: Color range: [MPEG]
[2023:06:02:00:23:13]: Error: Couldn't add Gamepad to ViGEm connection [E0000007]
[2023:06:02:00:23:14]: Error: Couldn't add Gamepad to ViGEm connection [E0000007]
[2023:06:02:00:23:15]: Error: Couldn't add Gamepad to ViGEm connection [E0000007]
[2023:06:02:00:23:16]: Error: Couldn't add Gamepad to ViGEm connection [E0000007]
[2023:06:02:00:23:17]: Error: Couldn't add Gamepad to ViGEm connection [E0000007]
[2023:06:02:00:23:48]: Info: CLIENT DISCONNECTED
[2023:06:02:00:23:58]: Info: Executing [Desktop]
[2023:06:02:00:23:58]: Info: CLIENT CONNECTED
[2023:06:02:00:23:58]: Info: 
Device Description : NVIDIA GeForce RTX 3070
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002484
Device Video Mem   : 8043 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 8154 MiB
Feature Level      : 0x0000B100
Capture size       : 1280x720
Offset             : 0x0
Virtual Desktop    : 1280x720
[2023:06:02:00:23:58]: Info: Desktop resolution [1280x720]
[2023:06:02:00:23:58]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:06:02:00:23:58]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.639648,0.339844]
Green Primary      : [0.299805,0.69043]
Blue Primary       : [0.137695,0.0380859]
White Point        : [0.282227,0.296875]
Min Luminance      : 0.01 nits
Max Luminance      : 1499 nits
Max Full Luminance : 799 nits
[2023:06:02:00:23:58]: Info: Capture format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:06:02:00:23:58]: Info: SDR color coding [Rec. 601]
[2023:06:02:00:23:58]: Info: Color range: [MPEG]
[2023:06:02:00:24:00]: Error: Couldn't add Gamepad to ViGEm connection [E0000007]
[2023:06:02:00:24:01]: Error: Couldn't add Gamepad to ViGEm connection [E0000007]
[2023:06:02:00:24:09]: Info: CLIENT DISCONNECTED
[2023:06:02:00:25:03]: Info: Executing [Desktop]
[2023:06:02:00:25:03]: Info: CLIENT CONNECTED
[2023:06:02:00:25:03]: Info: 
Device Description : NVIDIA GeForce RTX 3070
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002484
Device Video Mem   : 8043 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 8154 MiB
Feature Level      : 0x0000B100
Capture size       : 1280x720
Offset             : 0x0
Virtual Desktop    : 1280x720
[2023:06:02:00:25:03]: Info: Desktop resolution [1280x720]
[2023:06:02:00:25:03]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:06:02:00:25:03]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.639648,0.339844]
Green Primary      : [0.299805,0.69043]
Blue Primary       : [0.137695,0.0380859]
White Point        : [0.282227,0.296875]
Min Luminance      : 0.01 nits
Max Luminance      : 1499 nits
Max Full Luminance : 799 nits
[2023:06:02:00:25:03]: Info: Capture format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:06:02:00:25:03]: Info: SDR color coding [Rec. 601]
[2023:06:02:00:25:03]: Info: Color range: [MPEG]
[2023:06:02:00:25:13]: Info: 
Device Description : NVIDIA GeForce RTX 3070
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002484
Device Video Mem   : 8043 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 8154 MiB
Feature Level      : 0x0000B100
Capture size       : 1280x720
Offset             : 0x0
Virtual Desktop    : 1280x720
[2023:06:02:00:25:13]: Info: Desktop resolution [1280x720]
[2023:06:02:00:25:13]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:06:02:00:25:13]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.639648,0.339844]
Green Primary      : [0.299805,0.69043]
Blue Primary       : [0.137695,0.0380859]
White Point        : [0.282227,0.296875]
Min Luminance      : 0.01 nits
Max Luminance      : 1499 nits
Max Full Luminance : 799 nits
[2023:06:02:00:25:13]: Info: Capture format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:06:02:00:25:13]: Info: SDR color coding [Rec. 601]
[2023:06:02:00:25:13]: Info: Color range: [MPEG]
[2023:06:02:00:25:18]: Error: Couldn't add Gamepad to ViGEm connection [E0000007]
[2023:06:02:00:25:19]: Error: Couldn't add Gamepad to ViGEm connection [E0000007]
[2023:06:02:00:25:20]: Error: Couldn't add Gamepad to ViGEm connection [E0000007]
[2023:06:02:00:25:20]: Info: CLIENT DISCONNECTED
[2023:06:02:00:25:21]: Error: Couldn't add Gamepad to ViGEm connection [E0000007]
```

**Comment by mag2352** [comment]
I also have this same issue. Prevents me from playing games on the Microsoft Store and Epic Games Store, like Halo and Fortnite. The very first time initiating a streaming session from Moonlight on iOS, the controller works as intended. Every subsequent attempt at starting a session causes a “new” controller to be connected, and this controller is not detected by these games, even when closing and reopening them. Rebooting the computer is the only way to resolve this issue.

**Comment by ozboss** [comment]
Can confirm, please don't close

**Comment by RaghavRao** [comment]
Can also confirm, is preventing me from playing Halo, same issue.

As a workaround @mag2352 set "Force Gamepad #1" on Moonlight config. This will force ALL controllers to be player 1. This obviously breaks split screen co-op.

I'm working on seeing if HidHide from the ViGEmBus developer to hide some of the controllers, maybe eventually work on a PR to create a filter list of which controllers to show

**Comment by Stumpfhoelzern** [comment]
I have a similar problem, only that my controller doesn't work at all. With every button press or axis movement, the host starts to stutter badly, sometimes until I unplug the controller, and additional controllers appear in the Device Manager, all of them showing that no suitable driver is found/installed. Sunshine recognizes it correctly as an 360 controller, but than only generates multiple "Error: Couldn't add Gamepad to ViGEm connection [E0000007]". Reinstalling ViGEm, Sunshine, Moonlight didn’t help. 
Wasn't sure if this warrants an issue if its own.

**Comment by cyberweasel** [comment]
Also having issue were virtual controller stays in Device manager.   upon subsequent sessions, a second controller is added.  GamePass games seem to handle this fine (or at least the ones I play).  Epic Games do not.  They see the remote session as a second player.  Disconnecting the session and then deleting the ghost controller in DM resolves for next session but then it sticks again.
Windows 10 Host
Android client with USB-C controller

**Comment by cgutman** [comment]
Please try the latest nightly build: https://github.com/LizardByte/Sunshine/releases/nightly-dev

This was probably fixed by #1975

**Comment by cyberweasel** [comment]
> Please try the latest nightly build: https://github.com/LizardByte/Sunshine/releases/nightly-dev
> 
> This was probably fixed by #1975

Thank you.  I have upgraded to the nightly builds and so far this seems to resolve my issue.
